### PR TITLE
Engine: refactor PartialTransaction context

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
This is a preliminary work for #5500 to be able to reinterpret transaction node without transaction seed.

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [X] Include appropriate tests
- [X] Set a descriptive title and thorough description
- [X] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala [hunk 1]
```diff
@@ -311,7 +312,8 @@ object Speedy {
         checkSubmitterInMaintainers: Boolean,
         compiledPackages: CompiledPackages,
         scenario: Boolean,
-        seedWithTime: Option[(crypto.Hash, Time.Timestamp)] = None,
+        submissionTime: Time.Timestamp = Time.Timestamp.now(),
```
Comment (gerolf-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala:315]: Please remove this default value and rather enforce that we pass the right value in.

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala [hunk 2]
```diff
@@ -22,31 +22,42 @@ object PartialTransaction {
...
       val exeContext: Option[ExercisesContext], // empty if root context
       val children: BackStack[Value.NodeId],
-      protected val childrenSeeds: Stream[crypto.Hash]
+      protected val childrenSeeds: Int => Option[crypto.Hash]
   ) {
     def addChild(child: Value.NodeId): Context =
-      new Context(exeContext, children :+ child, childrenSeeds.drop(1))
-    def nextChildrenSeed: Option[crypto.Hash] = childrenSeeds.headOption
+      new Context(exeContext, children :+ child, childrenSeeds)
+    def nextChildrenSeed: Option[crypto.Hash] =
```
Comment (gerolf-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala:29]: As already before this PR, there is an assumption that `addChild` and `nextChildrenSeed` are always called in that order, right?

Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala:29]: First you build your node using `nextChildrenSeed` if this node require a seed (Create and Exercice) and then you add this node with `addChild`

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala]: Done.

**Pull request merged**